modules/bullet/collision_object_bullet.h

```diff
@@ -44,13 +44,35 @@ class btCollisionShape;
 
 class CollisionObjectBullet : public ShapeOwnerBullet {
 protected:
-	class HackedBasis : public Basis {
+	class EnhancedBasis : public Basis {
 	public:
 		void get_ABS_scale(Vector3 &out_vec) const {
 			out_vec[0] = Vector3(elements[0][0], elements[1][0], elements[2][0]).length();
 			out_vec[1] = Vector3(elements[0][1], elements[1][1], elements[2][1]).length();
 			out_vec[2] = Vector3(elements[0][2], elements[1][2], elements[2][2]).length();
 		}
+
+		real_t get_uniform_scale() const {
+			Vector3 scale;
+			get_ABS_scale(scale);
+			real_t uniform_scale = get_biggest_axis(scale);
```

Comment:
So why would you like to choose the axis with largest scaling?

Reply:
When a non-uniform scale is submitted I think the right  thing to do is to take the biggest axis as uniform scale, and show a warning. However we can use what we think it's right since the non-uniform scale is not intended to work.

Reply:
In my opinion, it'd be better to do something that emphasizes this problem is better, because people might miss a slightly wrong physics, but they won't miss a totally wrong physics (because I think we want to ensure that people avoids wrong physics).